### PR TITLE
merge similar procs regarding digits

### PR DESCRIPTION
**Pull request by ringabout**
- add `std/private/digitsutils`
- continue https://github.com/nim-lang/Nim/pull/16194

### CODE REVIEW

**Comment by Araq** [comment]
Great idea, please continue! :-)

File: lib/system/digitsutils.nim [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const digitsTable* = "0001020304050607080910111213141516171819" &
```
Comment (timotheecour) [inline lib/system/digitsutils.nim]: how about lib/std/private/digitsutils.nim instead?

Comment (ringabout) [reply lib/system/digitsutils.nim]: Sounds good.

File: lib/system/dragonbox.nim [hunk 2]
```diff
@@ -1039,19 +1043,6 @@ proc toDecimal64*(ieeeSignificand: uint64; ieeeExponent: uint64): FloatingDecima
 ## ==================================================================================================
 
 proc utoa2Digits*(buf: var openArray[char]; pos: int; digits: uint32) {.inline.} =
-  const
```
Comment (Araq) [inline lib/system/dragonbox.nim:1042]: This is also duplicated in schubfach.nim

Comment (ringabout) [reply lib/system/dragonbox.nim:1042]: I see

File: lib/system/dollars.nim [hunk 3]
```diff
@@ -6,22 +9,14 @@ proc `$`*(x: int): string {.magic: "IntToStr", noSideEffect.}
...
-      inc i
-      n = nn
-    result.setLen i
-
-    let half = i div 2
-    # Reverse
-    for t in 0 .. half-1: swap(result[t], result[i-t-1])
+    return "0"
+  elif x == 1:
+    return "1"
```
Comment (timotheecour) [inline lib/system/dollars.nim:14]: @xflywind why special case 1? or even 0 now that that the implementation changed?

Comment (timotheecour) [reply lib/system/dollars.nim:14]: => https://github.com/nim-lang/Nim/pull/18330

**Pull request merged**